src/main/java/org/passay/PasswordGenerator.java

```diff
@@ -17,10 +17,15 @@
  */
 public class PasswordGenerator
 {
+  /** Retry password generation at most this many times. */
+  private static final int RETRY_LIMIT = 2;
```

Comment:
I think this will need to be configurable. It's a fine default, but ultimately something that deployers will want to tweak.

Reply:
Fair enough, I'm fine with a wait and see approach here.

Reply:
@serac I would like to have it configurable too. It might be a range from 1 to ... 10? Something similar.  I would rather

Reply:
This is what I did in Apache Cassandra:

https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/db/guardrails/CassandraPasswordGenerator.java#L57-L77

The default is 100 attempts :D Yeah I know, this is an absolute overkill and some figure like "10" would be probably enough but I do not think that having the number of attempts so big is problematic because it will generate a valid password in 99% in two attempts anyway. That configuration property is not something a users are going to use frequently if ever.

At the same time, you do not have any real influence over the probability that the password will not be invalid. You may have a super bad luck and it would need to generate it three times, who knows? If you generate passwords to be 30 characters long, the probability of generating illegal sequence just raises. I would rather generate it by more attempts rather then failed completely.